compatibility_server/pip_checker.py

```diff
@@ -27,29 +27,54 @@
 import enum
 import json
 import logging
-import os.path
 import shlex
-import subprocess
-import tempfile
+import socket
 import urllib.request
 
 from typing import Any, List, Mapping, Optional
 
+import docker
+
+socket.setdefaulttimeout(3)
+
 PYPI_URL = 'https://pypi.org/pypi/'
+CONTAINER_WITH_PKG = "checker"
+TIME_OUT = 300  # seconds
+
+
+class PipCheckerError(Exception):
+    """Pip checker failed in an unexpected way."""
 
+    def __init__(self, error_msg: str):
+        super(PipCheckerError, self).__init__(
+            'Pip Checker failed, '
+            'logs are: {error_msg}'.format(error_msg=error_msg))
 
-class PipError(Exception):
+        # Convert error_msg to bytes as this needed to be returned in the
+        # response which should be bytes. And the message returned from docker
+        # is already bytes so that won't need the conversion.
+        if isinstance(error_msg, str):
+            self.error_msg = error_msg.encode('utf-8')
+        else:
+            self.error_msg = error_msg
```

Comment:
Having an exception with an error message in bytes in odd.  You should convert from str to bytes at the HTTP boundary.

Reply:
Removed the conversion.